### PR TITLE
Categorical

### PR DESCRIPTION
**Pull request by dmbates**
Too many changes in this commit but it is something to work with.

### CODE REVIEW

File: src/remat.jl [hunk 1]
```diff
@@ -66,19 +66,25 @@ end
 
 Base.eltype(R::ReMat) = eltype(R.z)
 
-function Base.copy!{S,R}(d::PooledDataVector{S,R}, s::PooledDataVector{S,R})
-    copy!(d.pool, s.pool)
+function Base.copy!(d::CategoricalVector, s::CategoricalVector)
```
Comment (nalimilan) [inline src/remat.jl]: Please define this function in CategoricalArrays.jl instead.

**Comment by dmbates** [comment]
Failures on AppVeyor are related to DataArrays on nightlies branch.

File: src/remat.jl (hunk 1 above)
Comment (nalimilan) [inline src/remat.jl]: Now in 0.1.0 (https://github.com/JuliaLang/METADATA.jl/pull/6792).

Comment (dmbates) [inline src/remat.jl]: @nalimilan Thanks for the reminder.  I'll delete my version from the master branch of MixedModels.

It turns out that I may not actually call that method.  I need to do some more checking.

**Pull request merged**